### ISSUE TITLE
About using self-attention

### ISSUE THREAD

**Issue by cmyyy** [lyndonzheng/Pluralistic-Inpainting#26]
Hello, i also use self-attention layer in my model, and some results look like this, where there are strange textures in the face.Have u ever met this situation and any advice on dealing with this? THX!~
ps:trained on celeba_hq, rectangle masks.
![](https://user-images.githubusercontent.com/26576058/62006117-f9625200-b16e-11e9-9f1e-cce680be8f2b.png)

![](https://user-images.githubusercontent.com/26576058/62006620-91fbd080-b175-11e9-9208-ff56d4c9a5ae.png)

**Comment by lyndonzheng** [comment]
@cmyyy I havenot met this situation for self-attention layer. In my model, the problem comes from the discriminator is not so stable. How do you know this strange textures come from the self-attention layer? Did you try to remove this layer in the model?

**Comment by cmyyy** [comment]
After i remove this layer, strange textures still exist, but may not that obvious.
By the way, not all images look like this even i use SA-layer, so i think this problem comes from 
 what you mentioned,unstable D. 
So, could you share any experience on building a stable D? And what's the improvement after adding SA-layer to D?

**Comment by lyndonzheng** [comment]
@cmyyy For this strange textures, some people suggest to use the nn.Upsample rather than the nn.ConvTranspose2d, such as the laest bigGAN and GrowingGAN. Maybe this is useful for your project. 
For the center/regular mask inpainiting, our model does not have very strange images. But for the free-from mask, some images are like that. Hence, our problem comes from the hard task, rather then the SA-layer,
As for the SA-layer in the D, this is important for our model. It can ensure the generated pixels hold global consistency semantic information to the visible part.

**Comment by cmyyy** [comment]
Thanks for your advice！
I think maybe the strange texture exists because my D cannot distinguish it well。As i use global and local D proposed by Jiahui Yu，i think i could try patchGAN or resnet+patchGAN+SA which seems more powerful.What do u think？：）



------------------ Original ------------------

**Comment by lyndonzheng** [comment]
@cmyyy For regular mask, the global and local D is suitable. The testure should be not so strange. For different GANs, different structures have different performance. In my experiment, the patchGAN is suitable for the local textures and SA-GAN is better for global semenatic consistency.  Besides, for your problem, the skin is not so realistic, do you think the TV smooth loss can ease them?

**Comment by cmyyy** [comment]
@lyndonzheng Thanks for your patience and helpful advice!
